feat(new bot): Added HypernymBot to the game

JoseManuelPintoLozano opened the pull request:
Added HypernymBot to the game
Imported in the main.py file to use it
In Board.py change the method to load the words file using utf8

JoseManuelPintoLozano committed:
Added HypernymBot to the game
Imported in the main.py file to use it
In Board.py change the method to load the words file using utf8

JoseManuelPintoLozano committed:
Added HypernymBot to the game
Imported in the main.py file to use it
In Board.py change the method to load the words file using utf8
Rebase branch into refactor to include pylint and the last changes

ikeinyyo commented on src/bots/HypernymBot.py:
```diff
@@ -0,0 +1,61 @@
+from .BotBase import BotBase
```
**S:** Sort imports

ikeinyyo commented on src/bots/HypernymBot.py:
```diff
@@ -0,0 +1,61 @@
+
+
+    def __get_first_common_hypernym(self, word1, word2, lang = 'spa'):
+        try:
```
I don't want to review this code. Please, refactor this code using better names, methods, and other techniques. Then we can talk.

ikeinyyo reviewed (changes requested):
Please, refactor the code

JoseManuelPintoLozano committed:
Refactor of the code started

JoseManuelPintoLozano committed:
Refactor of the code started

JoseManuelPintoLozano committed:
Solved endless loop in the refactor

JoseManuelPintoLozano committed:
Solved endless loop in the refactor

JoseManuelPintoLozano committed:
Reordered imports and created private dictionary

ikeinyyo committed:
feat(refactor): Add pylint and types annotations

Add Pylint

fix all pylint warnings

Add dataclass

Add types annotations

Fix pylint

Change name

JoseManuelPintoLozano committed:
Added types, done some refactor and a rebase from feature/refactor

JoseManuelPintoLozano committed:
Solved conflicts in main and board due to the names of the imports. Previously with uppercase now with lowercase

Commit:
Deleted class HypernymBot

Commit:
Solve one conflict in the Board.py file

ikeinyyo commented:
Fran approved this code
